VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionPostsBaseFragment.java

```diff
@@ -68,10 +59,26 @@ public void onItemClick(AdapterView<?> parent, View view, int position, long id)
                     thread.setRead(true);
                     discussionPostsAdapter.getView(position, view, parent);
                 }
-
-                CourseDiscussionPostsBaseFragment.this.onItemClick(thread, parent, view,
-                        position, id);
             }
         });
     }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+
+        if (isRestart) {
+        /*
+        If the activity/fragment needs to be reinstantiated upon restoration,
+        then in some cases the onRestart() callback maybe invoked before view
+        initialization, and thus the controller might not be initialized, and
+        therefore we need to guard this with a null check.
+         */
+            if (controller != null) {
+                nextPage = 1;
+                controller.resetSilently();
+            }
+        }
+        isRestart = true;
```

Comment:
`isStopped` is a more useful indicator overall in my opinion, as it can be used for other purposes than this exact one. `isRestart` is not useful for any other purpose, and in fact is a false indicator for most of the Fragment's lifecycle.

> Otherwise, the function will read like "if fragment `isStopped` then `do a silent refresh`" which isn't what we want to imply 😉 

That reads fine to me, since that's exactly what restart means; starting again from a stopped state.

----

Anyway, if you want to stay with this, then it should probably be renamed to `isRestarting` instead, which is at least correct grammatically 😜 . Though, as I mentioned, it would still be wrong semantically over the lifecycle of the Fragment.